src/openMVG/sfm/AlembicImporter.hpp

```diff
@@ -25,17 +20,14 @@ class AlembicImporter
 {
 public:
   explicit AlembicImporter(const std::string &filename);
-  ~AlembicImporter() = default;
+  ~AlembicImporter();
 
   void populate(sfm::SfM_Data &sfmdata, sfm::ESfM_Data flags_part = sfm::ESfM_Data::ALL);
 
 private:
-  bool readPointCloud(IObject iObj, M44d mat, sfm::SfM_Data &sfmdata, sfm::ESfM_Data flags_part);
-  bool readCamera(IObject iObj, M44d mat, sfm::SfM_Data &sfmdata, sfm::ESfM_Data flags_part, const index_t sampleFrame = 0);
   
-  //@todo complete the interface, also maybe parameters need to be passed by reference?
-  void visitObject(IObject iObj, M44d mat, sfm::SfM_Data &sfmdata, sfm::ESfM_Data flags_part = sfm::ESfM_Data::ALL);
-  IObject _rootEntity;
+  struct ObjImpl;
+  std::unique_ptr<ObjImpl> _objImpl;
```

Comment:
is there anything else to do in the cpp?
Because if I just do as u propose above then i get a lot of 
```
/Users/sgaspari/dev/code/popart/libs/openMVG/src/openMVG/sfm/AlembicImporter.cpp:82:21: error: reference to 'IObject' is ambiguous
bool readPointCloud(IObject iObj, M44d mat, sfm::SfM_Data &sfmdata, sfm::ESfM_Data flags_part)
                    ^
/Users/sgaspari/dev/code/popart/libs/openMVG/src/openMVG/sfm/AlembicImporter.hpp:18:7: note: candidate found by name lookup is 'Alembic::Abc::IObject'
class IObject;
      ^
/Users/sgaspari/dev/code/popart/libs/alembic-1.5.8/include/Alembic/Abc/IObject.h:52:7: note: candidate found by name lookup is 'Alembic::Abc::v7::IObject'
class IObject : public Base
```
like it thinks that the two classes are different. And then also
```
/Users/sgaspari/dev/code/popart/libs/openMVG/src/openMVG/sfm/AlembicImporter.cpp:375:22: error: allocation of incomplete type 'Alembic::Abc::IObject'
  _objImpl.reset(new IObject(archive.getTop()));
                     ^~~~~~~
/Users/sgaspari/dev/code/popart/libs/openMVG/src/openMVG/sfm/AlembicImporter.hpp:18:7: note: forward declaration of 'Alembic::Abc::IObject'
class IObject;
      ^
```